client/src/session.rs

```diff
@@ -149,8 +150,20 @@ impl Session {
         let last_seen = self.dispatcher.last_seen();
 
         if last_seen + expiration < Instant::now() {
+            // Sending 3 pings after each other to avoid lost UDP packets.
+            // We need only one response.
+            // Note: futures are asynchronous, because we shouldn't wait for ping timeout
+            //       in case of lost packets.
+            futures::future::select_ok((0..3).into_iter().map(|i| {
+                async move {
+                    tokio::time::delay_for(Duration::from_millis(200 * i)).await;
+                    self.ping().await
+                }
+                .boxed_local()
+            }))
```

Comment:
Suggestion for a follow up PR: I'd increase the duration a bit and break in case `last_seen` is updated (but still iterating 3 times)

Reply:
It will break after first future will return Ok value